File: flutter_ui/lib/assets/asset_plugin.dart
class AssetPlugin {
  Map<String, String>? _props;
  AssetPlugin();
  dynamic invoke(dynamic asset) => asset;
  void setProperty(String name, String value) {
    _props ??= {};
    _props![name] = value;
  }

  String? getProperty(String name, [String? defaultValue]) {
    if (_props == null) return defaultValue;
    return _props![name] ?? defaultValue;
  }
}

File: luaui/assets/AssetPlugin.lua
local AssetPlugin = {}
AssetPlugin.__index = AssetPlugin
function AssetPlugin.new()
    local self = setmetatable({}, AssetPlugin)
    self._props = {}
    return self
end
function AssetPlugin:invoke(asset) return asset end
function AssetPlugin:setProperty(name, value) self._props[name] = value end
function AssetPlugin:getProperty(name, defaultValue) return self._props[name] or defaultValue end
return AssetPlugin
